posts/process-csvs-from-amazon-s3-using-apache-flink-jhipster-and-kubernetes.md

```diff
@@ -0,0 +1,187 @@
+
+Now we can compile and start our JHipster application:
+
+```bash
+# Compile the application and the job
+./mvnw clean verify
+
+# Start the application
+./mvnw
+```
+
+Browse to [http://localhost:8080](http://localhost:8080) and then login using the credentials `admin/admin`. Using the top menu, navigate to `Administration -> API` and you should see the resource `covid-resource`:
+
+![API](https://raw.githubusercontent.com/Falydoor/blog-usa/blog-flink/images/2021/01/flink-2.png)
+
+After clicking on `Execute`, the job will be triggered and you will see the result after few seconds:
+
+![Result](https://raw.githubusercontent.com/Falydoor/blog-usa/blog-flink/images/2021/01/flink-3.png)
+
+Future work would be to build a small UI to show the results in a chart rather than just a json. Also, it should be possible to pick the date and the field to sort on.
+
+# Job manager’s web UI
+
+What is pretty cool about Flink is that you have access to a pretty [neat dashboard](http://localhost:8081/). The screenshot below shows all the different stages on the job we just ran:
+
+![Dashboard](https://raw.githubusercontent.com/Falydoor/blog-usa/blog-flink/images/2021/01/flink-4.png)
+
+We can see all the tasks executed and their duration, also, the diagram is all interactive which makes things very easy to navigate in case of a complex job.
+
+# Data streaming
+
+The [DataStream API](https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/datastream_api.html) will let you do all the regular operations on your stream (filtering, updating state, defining windows, aggregating, etc...). Flink provides connectors to interface with the regular streaming services like [Apache Kafka](https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/connectors/kafka.html) and [Amazon Kinesis](https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/connectors/kinesis.html).
+
+An other nice feature available with Flink is its [Event Processing library](https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/libs/cep.html) that will detect event patterns so you can focus on what is really important in your event data.
+
+# To sum up
+
+Apache Flink released its first API-stable version in March 2016 and process data in-memory just like Spark. The big advantage of Flink is its stream processing engine that can also do batch processing. In addition, Flink does very well with memory management and speed ([more details on Flink vs Spark](https://www.dezyre.com/article/apache-flink-vs-spark-will-one-overtake-the-other/282)).
+
+I recommend opting for Flink as it will do most of the things that Spark does while being way more efficient at processing data streams. Additionally, Flink can be deployed on most common resource providers like Kubernetes and also as stand-alone cluster on bare-metal hardware. For more real-world use cases and detailed explanations, please [read this page](https://flink.apache.org/usecases.html) from the Flink's website.
```

Comment:
```suggestion
I recommend opting for Flink as it will do most of the things that Spark does while being way more efficient at processing data streams. Additionally, Flink can be deployed on most common resource providers like Kubernetes and also as a stand-alone cluster on bare-metal hardware. For more real-world use cases and detailed explanations, please [read this page](https://flink.apache.org/usecases.html) from the Flink's website.
```